xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/BaseObject.java

```diff
@@ -366,7 +366,7 @@ public void set(String fieldname, java.lang.Object value, XWikiContext context)
         PropertyClass pclass = (PropertyClass) bclass.get(fieldname);
         BaseProperty prop = (BaseProperty) safeget(fieldname);
         if ((value instanceof String) && (pclass != null)) {
-            prop = pclass.fromString((String) value);
+            prop = pclass.fromString((String) value, prop);
```

Comment:
I'm not sure it's really the cleanest code in general, it's good to reuse the property, but it's a bit strange to re-put it in the object.